### PR TITLE
Re-write "Scaling Out Airflow" Guide

### PR DESCRIPTION
**Pull request by jwitz**
The Scaling Airflow guide, one of the most commonly visited guides, was not in great shape. So far this PR only encompasses wording and style changes, but I'm hoping to add more technical details or updates as needed. 

I took out most of the references to Astronomer because the screenshots were pre 0.13, but definitely want to reintroduce them at some point. Interested to hear thoughts on how we should mention Astronomer features in Airflow guides, as it doesn't seem like what we had before was best practice.

### CODE REVIEW

**Comment by paolaperaza** [comment]
This is such an overdue set of changes - thanks a ton for getting this PR started, @jwitz. I ended up going in here to add some content and restructure the headers to fit that new content. Would love a second pair of eyes on those changes, and I think everything is ready to be reviewed from a technical standpoint.

@kentdanas Any chance you have a few mins to give this a glance and let me know if it's technically sound/accurate? Happy to hop back in here for comments/feedback. Much appreesh 🙏

File: guides/airflow-scaling-workers.md [hunk 1]
```diff
@@ -41,50 +58,90 @@ Here are the settings and their default values.
...
+
+Let's assume we have an Airflow environment that uses the default settings defined above and runs 4 Celery Workers. If we have 4 Celery Workers and `worker_concurrency=16`, we could theoretically run 64 tasks at once. Because `parallelism=32`, however, only 32 tasks are able to run at once across Airflow.
+
+If all of these tasks exist within a single DAG and `dag_concurrency=16`, however, we'd be further limited to a maximum of 16 tasks at once.
+
+## DAG-level Airflow Settings
+
+There are two primary DAG-level Airflow settings users can define in code:
+
+- **`max_active_run`** is the maximum number of active DAG Runs allowed for the DAG in question. Once this limit is hit, the Scheduler will not create new active DAG Runs. If this setting is not defined, the value of `max_active_runs_per_dag` (described above) is assumed.
```
Comment (kentdanas) [inline guides/airflow-scaling-workers.md:95]:
```suggestion
- **`max_active_runs`** is the maximum number of active DAG Runs allowed for the DAG in question. Once this limit is hit, the Scheduler will not create new active DAG Runs. If this setting is not defined, the value of `max_active_runs_per_dag` (described above) is assumed.
```

**Review by kentdanas — APPROVED** [review]
@paolaperaza this looks really good! Sooo much better than before. Agree with @jwitz that we could add some info back in at some point on managing this on Astronomer; fwiw in some of my guides I've added a separate section for this so that it doesn't 'take over' the main guide, and I think something similar could make sense here (even a brief section on how to use these Airflow configurations concurrently with resource settings for the deployment might be nice). 

I made one suggestion with a very minor correction, but other than that it looks good to me!

**Comment by paolaperaza** [comment]
Sweet - thank you for the review @kentdanas! I just added a few more references to Astro docs. It feels like the opportunity to talk about Astronomer functionality would primarily focus on:

- Setting env vars
- Configuring resources

I _feel_ like just linking out to docs on those things is fine? Or do you think there's room to add even more before we merge this?

**Comment by kentdanas** [comment]
@paolaperaza the links you added seem good to me! I can't really think of too much other astro-specific content we would want to call out.

File: guides/airflow-scaling-workers.md [hunk 2]
```diff
@@ -7,13 +7,30 @@ heroImagePath: null
 tags: ["Workers", "Concurrency", "Parallelism", "DAGs"]
 ---
 <!-- markdownlint-disable-file -->
-As data pipelines grow in complexity, the need to have a flexible and scalable architecture is more important than ever. Airflow's Celery Executor makes it easy to scale out workers horizontally when you need to execute lots of tasks in parallel. There are however some "gotchas" to look out for. Even folks familiar with using the Celery Executor might wonder, "Why are more tasks not running even after I add workers?""
 
-This common issue is the result of a number of different Airflow settings you can tweak to reach max parallelism and performance. Settings are maintained in the airflow.cfg file and can also be overridden using environment variables.
+## Overview
 
-See the guide on [Airflow Executors](https://www.astronomer.io/guides/airflow-executors-explained/) for more info on which executor is right for you.
+One of Apache Airflow's biggest strengths is its ability to scale as the needs of your team grow. To make the most of Airflow's extensibility, there are a few key settings that you should consider modifying as you scale up your data pipelines.
```
Comment (jwitz) [inline guides/airflow-scaling-workers.md:13]:
```suggestion
One of Apache Airflow's biggest strengths is its scalability. To make the most of Airflow, there are a few key settings that you should consider modifying as you scale up your data pipelines.
```

File: guides/airflow-scaling-workers.md [hunk 3]
```diff
@@ -7,13 +7,30 @@ heroImagePath: null
...
+
+Environment-level settings are generally set in the `airflow.cfg` file or through environment variables, while DAG and task-level settings are typically specified in code.
+
+In this guide, we'll walk through key values in each category in the context of scale and performance.
+
+## Environment-level Airflow Settings
+
+Environment-level settings are typically set in the `airflow.cfg` file, which is required in all Airflow projects. All settings in this file have a default value that can be overriden by modifying the file directly. Default values are taken from the [default_airflow.cfg file](https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg) in Airflow project source code.
+
+To check current values for an existing Airflow environment, navigate to **Admin** > **Configurations** in the Airflow UI. For more information, read [Setting Configuration Options](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html) or [Configuration Reference](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html) in Airflow documentation.
```
Comment (jwitz) [inline guides/airflow-scaling-workers.md:29]:
```suggestion
To check current values for an existing Airflow environment, go to **Admin** > **Configurations** in the Airflow UI. For more information, read [Setting Configuration Options](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html) or [Configuration Reference](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html) in Airflow documentation.
```

File: guides/airflow-scaling-workers.md [hunk 4]
```diff
@@ -7,13 +7,30 @@ heroImagePath: null
...
+- DAG-level settings
+- Operator-level settings
+
+Environment-level settings are generally set in the `airflow.cfg` file or through environment variables, while DAG and task-level settings are typically specified in code.
+
+In this guide, we'll walk through key values in each category in the context of scale and performance.
+
+## Environment-level Airflow Settings
+
+Environment-level settings are typically set in the `airflow.cfg` file, which is required in all Airflow projects. All settings in this file have a default value that can be overriden by modifying the file directly. Default values are taken from the [default_airflow.cfg file](https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg) in Airflow project source code.
```
Comment (jwitz) [inline guides/airflow-scaling-workers.md:27]:
```suggestion
Environment-level settings are typically set in the `airflow.cfg` file, which is required in all Airflow projects. All settings in this file have a default value that can be overridden by modifying the file directly. Default values are taken from the [default_airflow.cfg file](https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg) in Airflow project source code.
```

File: guides/airflow-scaling-workers.md [hunk 5]
```diff
@@ -41,50 +58,92 @@ Here are the settings and their default values.
...
 
-And then assign your task to the pool
-```python
-t1 = MyOperator(
-  task_id='pull_from_api',
-  dag=dag,
-  pool='My_REST_API'
-)
-```
+> **Note:** If you're developing locally with the Astronomer CLI, you can define Airflow Pools in the `airflow_settings.yaml` file that was automatically generated when you initialized your Airflow project with `$ astro dev init`. This file allow you to set Pools, Variables, and Connections in a single file such that you don't have to manually re-create them in the Airflow UI every time you restart your environment. For more information, read [Customize your Airflow Image on Astronomer](https://www.astronomer.io/docs/cloud/stable/develop/customize-image#configure-airflowsettingsyaml).
```
Comment (jwitz) [inline guides/airflow-scaling-workers.md:149]:
```suggestion
> **Note:** If you're developing locally with the Astronomer CLI, you can define Airflow Pools in the `airflow_settings.yaml` file that was automatically generated when you initialized your Airflow project with `$ astro dev init`. You can use this file to set Pools, Variables, and Connections without needing to manually recreate them in the Airflow UI whenever you restart your environment. For more information, read [Customize your Airflow Image on Astronomer](https://www.astronomer.io/docs/cloud/stable/develop/customize-image#configure-airflowsettingsyaml).
```

**Comment by paolaperaza** [comment]
Sweet, thanks again for the review here @kentdanas and for the final additions @jwitz ! Merging 😊

**Pull request merged**